### DISCUSSION TITLE
Sending webhooks

### DISCUSSION THREAD

**Discussion by rivalsystems** [Budibase/budibase#4958] [Ideas]
It would be nice if an automation action was available to **send** a webhook.

Basically i could make an application that can trigger external jobs on my other 3rd party apps.

**Comment by mjashanks** [comment]
You can do this already!

1, Setup a new REST data source
2. Add your rest query
3. IN your automation, add an "External Data connector" and choose your REST query

**Reply by havedill** [reply]
Is there a way to pass parameters from my application into the webhook as well? Sorry for the questions i didnt find any documentation about this specific use case

**Reply by mjashanks** [reply]
Yes - this describes it: https://docs.budibase.com/docs/rest-bindings

**Reply by havedill** [reply]
Thank you, i've gotten this to work now. However - It seems like my webhook triggers a bunch while im editing my app, even though im not pressing the submit button. Any reason it would be doing that? Happens when i load the published app as well, versus pressing the submit button

**Reply by mjashanks** [reply]
If you are using it in a datasource, it will trigger the webhook in order to display the data in the preview. A Data Provider is really meant for `GET` requests.

You can also trigger your webhook using a button action:
1. Create a button
2. Click `Define Actions`
3. Add `Execute query` 

If you need to display the result of the request, then you can add an `Update State` action to save the response. Then, you can use the `State.YourStateKey` in bindings on the screen.
